hierarchicalforecast/methods.py

```diff
@@ -59,23 +64,243 @@ def _reconcile(S: np.ndarray, P: np.ndarray, W: np.ndarray,
             res[f'hi-{lv}'] = res['mean'] + zs * sigmah
     return res
 
+# %% ../nbs/methods.ipynb 5
+class Normality:
+    
+    def __init__(
+            self, 
+            sigmah: np.ndarray #Estimate of the standard deviation of the h-step forecast of size (`base`, `horizon`)
+        ):
+        self.sigmah = sigmah
+        
+    def get_samples(self):
+        return self.sigmah
+
 # %% ../nbs/methods.ipynb 6
-def bottom_up(S: np.ndarray,
-              y_hat: np.ndarray,
-              idx_bottom: List[int],
-              sigmah: Optional[np.ndarray] = None, 
-              level: Optional[List[int]] = None,
-              bootstrap: bool = False,
-              bootstrap_samples: Optional[np.ndarray] = None):
-    n_hiers, n_bottom = S.shape
-    P = np.zeros_like(S, dtype=np.float32)
-    P[idx_bottom] = S[idx_bottom]
-    P = P.T
-    W = np.eye(n_hiers, dtype=np.float32)
-    return _reconcile(S, P, W, y_hat, sigmah=sigmah, level=level, 
-                      bootstrap=bootstrap, bootstrap_samples=bootstrap_samples)
+class Bootstrap:
+    
```

Comment:
It would be good to add references to Gamakumara's work for this class, in its docstring.

Reply:
Additionally, the description of the algorithm in similar fashion to that of PERMBU would be of interest to our power users.

Reply:
Lets also add some warning to PERMBU and Bootstrap on the importance of not overfitted residuals.